cytopipe/cli/cmd.py

```diff
@@ -27,59 +27,76 @@ def run_cmd() -> None:
     -------
     None
     """
-    # checking provided cli arguments
-    cli_check(args_list=sys.argv)
+    # create args handler
+    # -- Cli Control Panel
+    args_handler = CliControlPanel(sys.argv)
+    # print(args_handler.__repr__())
 
-    # collecting parameters
-    params = sys.argv[1:]
-    mode_type = params[0]
-
-    # parsing arguments based on modes
-    if mode_type == "init":
+    # checking is user wanted to cli help
+    if args_handler.cli_help is True:
+        print(cli_docs)
+        sys.exit(0)
 
-        # parsing inputs
-        init_args = parse_init_args(params)
+    match args_handler.mode:
 
-        # setting up file paths
-        barcode_path = str(Path(init_args.barcode).absolute())
-        metadata_path = str(Path(init_args.metadata).absolute())
+        case "init":
 
-        # create data folder
-        # raises error if data directory exists (prevents overwriting)
-        data_dir_obj = Path("data")
-        data_dir_obj.mkdir(exist_ok=False)
-        data_dir_path = str(data_dir_obj.absolute())
+            # if mode help flag is shown, show cli help doc
+            if args_handler.mode_help is True:
+                print(init_doc)
+                sys.exit(0)
 
-        # moving all input files
-        for data_file in init_args.data:
-            f_path = str(Path(data_file).absolute())
-            shutil.move(f_path, data_dir_path)
+            # setting up input files for cytopipe
+            print("INFO: Formatting input files")
+            init_args = args_handler.parse_init_args()
 
-        shutil.move(barcode_path, data_dir_path)
-        shutil.move(metadata_path, data_dir_path)
+            # setting up file paths
+            barcode_path = str(Path(init_args.barcode).absolute())
+            metadata_path = str(Path(init_args.metadata).absolute())
 
-    elif mode_type == "run":
+            # create data folder
+            # raises error if data directory exists (prevents overwriting)
+            data_dir_obj = Path("data")
+            data_dir_obj.mkdir(exist_ok=False)
+            data_dir_path = str(data_dir_obj.absolute())
 
-        # selecting workflow process
-        proc_sel = params[1]
+            # moving all input files
+            for data_file in init_args.data:
+                f_path = str(Path(data_file).absolute())
+                shutil.move(f_path, data_dir_path)
 
-        # executing process
-        if proc_sel == "cp_process":
-            cp_process_args = parse_cp_process_args(params)
+            shutil.move(barcode_path, data_dir_path)
+            shutil.move(metadata_path, data_dir_path)
```

Comment:
Can you remind me what this is doing? A user will specify a path to these files and we'll move them to where we expect them to be?

Can you add a comment somewhere clarifying this logic?

Reply:
> Can you add a comment somewhere clarifying this logic?

of course! comments added on commit: c43041d439ed912a64a589ab61774511fe76dfb3

>Can you remind me what this is doing? A user will specify a path to these files and we'll move them to where we expect them to be?

Correct, a user will provide paths of the required files. Therefore when calling `cytopipe init`, it will create a `data` folder on the working directory containing those files.